pkg/backporter/backporter.go

```diff
@@ -278,94 +304,218 @@ func GetBugHandler(client bugzilla.Client) HandlerFuncWithErrorReturn {
+		PRs:          prs,
+		CloneTargets: targetVersions.List(),
+	}
+	return &wrpr, nil
+}
+
+// ClonesHandler acts as a router for the RESTish calls to clones
+func ClonesHandler(client bugzilla.Client, allTargetVersions sets.String) HandlerFuncWithErrorReturn {
+	return func(w http.ResponseWriter, r *http.Request) error {
+		if r.Method != "GET" {
+			w.WriteHeader(http.StatusBadRequest)
+			wpErr := writePage(w, "Error.", errorTemplate, fmt.Sprintf("%s - query incorrect", BugIDQuery))
+			if wpErr != nil {
+				http.Error(w, "Error building page.", http.StatusInternalServerError)
+				return wpErr
+			}
+			return fmt.Errorf("Bad request - clones can handle only GET requests")
+		}
+		return GetClonesHandler(client, allTargetVersions, w, r)
+	}
+}
+
 // GetClonesHandler returns an HTML page with detais about the bug and its clones
-func GetClonesHandler(client bugzilla.Client) HandlerFuncWithErrorReturn {
+func GetClonesHandler(client bugzilla.Client, allTargetVersions sets.String, w http.ResponseWriter, req *http.Request) error {
+	bugIDStr := req.URL.Query().Get(BugIDQuery)
+	if bugIDStr == "" {
+		w.WriteHeader(http.StatusBadRequest)
+		wpErr := writePage(w, "Error.", errorTemplate, fmt.Sprintf("%s - query incorrect", BugIDQuery))
+		if wpErr != nil {
+			http.Error(w, "Error building page.", http.StatusInternalServerError)
+			return wpErr
+		}
+		return fmt.Errorf("%s - query incorrect", BugIDQuery)
+	}
+	bugID, err := strconv.Atoi(bugIDStr)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		wpErr := writePage(w, "Error.", errorTemplate, fmt.Sprintf("Unable to parse bug id: %v", err))
+		if wpErr != nil {
+			http.Error(w, "Error building page.", http.StatusInternalServerError)
+			return wpErr
+		}
+		return err
+	}
+
+	wrpr, err := getClonesTemplateData(bugID, w, client, allTargetVersions)
+	if err != nil {
+		return err
+	}
+	wpErr := writePage(w, "Clones", clonesTemplate, wrpr)
+	if wpErr != nil {
+		http.Error(w, "Error building page.", http.StatusInternalServerError)
+		return wpErr
+	}
+	return nil
+}
+
+// CreateCloneHandler will create a clone of the specified ID and return success/error
+func CreateCloneHandler(client bugzilla.Client, allTargetVersions sets.String) HandlerFuncWithErrorReturn {
 	return func(w http.ResponseWriter, req *http.Request) error {
-		bugIDStr := req.URL.Query().Get(BugIDQuery)
-		if bugIDStr == "" {
+		if req.Method != "POST" {
 			w.WriteHeader(http.StatusBadRequest)
-			wpErr := writePage(w, "Error!", emptyTemplate, fmt.Sprintf("%s - query incorrect", BugIDQuery))
+			wpErr := writePage(w, "Error.", errorTemplate, fmt.Sprintf("%s - query incorrect", BugIDQuery))
 			if wpErr != nil {
-				http.Error(w, "Error building page!", http.StatusInternalServerError)
+				http.Error(w, "Error building page.", http.StatusInternalServerError)
 				return wpErr
 			}
-			return fmt.Errorf("%s - query incorrect", BugIDQuery)
+			return fmt.Errorf("Bad request - clones/create can handle only POST requests")
```

Comment:
error messages never start capitalized unless its a name of something by convention